specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json

```diff
@@ -453,6 +453,12 @@
           },
           {
             "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "name": "artifactType",
+            "in": "query",
+            "type": "string",
+            "description": "This sets the type of artifact being returned, when empty no artifact endpoint is returned."
           }
```

Comment:
send this in the payload not as a query parameter

Reply:
@rkmanda adding context the ask for change GET to POST for listclusterusercredentials API in AKS at the time came from ARM engineering team.

Reply:
The problem is not with the POST action but with having a query parameter. I dont quite follow the challenge with including the parameter in the request payload.

https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#:~:text=All%20parameters%20for%20the%20action%20request%20should%20be%20contained%20in%20the%20request%20body%20%E2%80%93%20and%20not%20in%20the%20URL.%20This%20matches%20the%20OData%20and%20Azure%20REST%20guidelines%20and%20will%20avoid%20impacting%20the%20authZ%20checks.

Reply:
Approving as on-off implementation. For future please follow the guideline as explained

Here are the drawbacks to putting something on the URI. For these reasons we strongly recommend placing the parameters in the body of the request. If we encounter a strong justification for why the pattern cannot be followed, we are open to providing an exception 

- You have to deal with URI encoding which could munge up the data
- Intermediary services may have to worry about sanitizing the data on the URI during logging.  This is easy for well known patterns like SAS URIs but not easy for arbitrary query params
- You won't be able to use linked access checks for any content in query params (but you could if the content is in the body)
- The URI can only hold so much content.  If you put something on the query string and then realize users need to pass multiples of it or the content becomes long you may be stuck with a big breaking change to pull it into the request body

Reply:
Thanks for the feedback and discussion offline